process-cmds.sh

```diff
@@ -94,6 +94,166 @@ process-iperf() {
+
+  # Wait to learn flows and hardware offload
+  sleep "${FLOW_LEARNING_TIME}"
+
+  # Record ethtool stats
+  echo "kubectl exec -it -n \"${FT_NAMESPACE}\" \"${TEST_TOOLS_POD}\" -- /bin/sh -c \"ethtool -S ${TEST_VF_REP} | sed -n 's/^\s\+//p'\""
+  ethtoolstart=$(kubectl exec -it -n "${FT_NAMESPACE}" "${TEST_TOOLS_POD}" -- /bin/sh -c "ethtool -S ${TEST_VF_REP} | sed -n 's/^\s\+//p'")
+  echo "${ethtoolstart}" >> "${HWOL_VALIDATION_FILENAME}"
+
+  # Record RX/TX packet counts
+  rxpktstart=$(echo "$ethtoolstart" | sed -n "s/^rx_packets:\s\+//p" | sed "s/[^0-9]//g")
+  txpktstart=$(echo "$ethtoolstart" | sed -n "s/^tx_packets:\s\+//p" | sed "s/[^0-9]//g")
+
+  # Start tcpdump
+  echo "kubectl exec -it -n \"${FT_NAMESPACE}\" \"${TEST_TOOLS_POD}\" -- /bin/sh -c \"timeout --preserve-status ${TCPDUMP_RUNTIME} tcpdump -i ${TEST_VF_REP} -n not arp\""
+  kubectl exec -it -n "${FT_NAMESPACE}" "${TEST_TOOLS_POD}" -- /bin/sh -c "timeout --preserve-status ${TCPDUMP_RUNTIME} tcpdump -i ${TEST_VF_REP} -n not arp" > "${TCPDUMP_FILENAME}"
+  cat "${TCPDUMP_FILENAME}" >> "${HWOL_VALIDATION_FILENAME}"
+
+  # Record ethtool stats
+  echo "kubectl exec -it -n \"${FT_NAMESPACE}\" \"${TEST_TOOLS_POD}\" -- /bin/sh -c \"ethtool -S ${TEST_VF_REP} | sed -n 's/^\s\+//p'\""
+  ethtoolend=$(kubectl exec -it -n "${FT_NAMESPACE}" "${TEST_TOOLS_POD}" -- /bin/sh -c "ethtool -S ${TEST_VF_REP} | sed -n 's/^\s\+//p'")
+  echo "${ethtoolend}" >> "${HWOL_VALIDATION_FILENAME}"
+
+  rxpktend=$(echo "$ethtoolend" | sed -n "s/^rx_packets:\s\+//p" | sed "s/[^0-9]//g")
+  txpktend=$(echo "$ethtoolend" | sed -n "s/^tx_packets:\s\+//p" | sed "s/[^0-9]//g")
+
+  rxcount=$(( rxpktend - rxpktstart ))
+  txcount=$(( txpktend - txpktstart ))
+
+  echo "Ethtool results for ${TEST_CLIENT_CLIENT_VF_REP}:"
+  echo "================================================="
+  echo "RX Packets: ${rxpktend} - ${rxpktstart} = ${rxcount}"
+  echo "TX Packets: ${txpktend} - ${txpktstart} = ${txcount}"
+
+  # Wait for Iperf to finish
+  wait $IPERF_PID
+
+  # Concatenate the background Iperf results into the same file
+  cat "${IPERF_FILENAME}" >> "${HWOL_VALIDATION_FILENAME}"
+
+  # Dump command output
+  if [ "$VERBOSE" == true ]; then
+    echo "Full Tcpdump Output (from ${HWOL_VALIDATION_FILENAME}):"
+    cat ${TCPDUMP_FILENAME}
+    echo "Full Iperf Output (from ${HWOL_VALIDATION_FILENAME}):"
+    cat ${IPERF_FILENAME}
+  else
+    echo "Summary Tcpdump Output (see ${HWOL_VALIDATION_FILENAME} for full detail):"
+    tail ${TCPDUMP_FILENAME}
+    echo "Summary Iperf Output (see ${HWOL_VALIDATION_FILENAME} for full detail):"
+    cat ${IPERF_FILENAME} | grep -B 1 -A 1 "sender"
+  fi
+
+  cat ${IPERF_FILENAME} | grep -cq "sender" && retVal=0 || retVal=1
+
+  if (( rxcount > THRESHOLD_PKT_COUNT )) || (( txcount > THRESHOLD_PKT_COUNT )); then
+    retVal=1
+  fi
+
+  # Cleanup temporary files
+  rm "${IPERF_FILENAME}"
+  rm "${TCPDUMP_FILENAME}"
+
+  return $retVal
+}
+
+process-hw-offload-validation() {
```

Comment:
Based on the README, there is a lot of output by this command. Once I get everything working, I tend to trim the output back to: 1) commands I need to rerun the test manually 2) SUCCESS or FAILED
For 1), yours is more complicated because of the waiting and reading of counters, so you will need to decide what is important to reproduce command manually, if it is possible. Also, you may not be able to silence some of what the commands spit out, but just looking at the README, there is a lot there. That is what the files are for. Use a combination of FT_DEBUG and VERBOSE to allow more output when needed.

Reply:
I trimmed the output. I used FT_DEBUG for some of the intermediate commands to determine the VF rep and other variables.